components/Button.tsx

```diff
@@ -0,0 +1,52 @@
+import * as React from 'react';
+import {
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  Image,
+  ImageSourcePropType,
+  TextStyle,
+  StyleProp,
+  ViewStyle,
+  GestureResponderEvent,
+} from 'react-native';
+
+const styles = StyleSheet.create({
+  buttonText: {
+    color: '#2c2c2c',
+    textAlign: 'left',
+    fontSize: 16,
+    fontWeight: 'bold',
+    textAlignVertical: 'center',
+  },
+  button: {
+    height: 50,
+    backgroundColor: 'white',
+    padding: 10,
+    flexDirection: 'row',
+  },
+  icon: {
+    width: 28,
+    height: 28,
+    marginLeft: 10,
+    marginRight: 30,
+  },
+});
+
+type Props = {
+  buttonStyle?: StyleProp<ViewStyle>;
+  onPress: (event?: GestureResponderEvent) => void;
+  iconSrc: ImageSourcePropType;
+  textStyle?: StyleProp<TextStyle>;
+};
+
+const Button: React.FC<Props> = ({ buttonStyle, onPress, iconSrc, textStyle, children }) => {
```

Comment:
```suggestion
const Button: React.FC<Props> = ({ buttonStyle = {}, onPress, iconSrc, textStyle = {}, children }) => {
```

Since both are optional I prefer to set a default value to these props instead of passing an `undefined` directly to the component.

Reply:
yep, that works. I think it's fine with undefined tho